docs/testplan/QoS-remapping-for-Tunnel-traffic-test-plan.md

```diff
@@ -0,0 +1,82 @@
+- [Test Cases](#test-cases)
+  - [Test case group for packet encapsulation](#test-case-group-for-packet-encapsulation)
+  - [Test case group for packet decapsulation](#test-case-group-for-packet-decapsulation)
+
+## Overview
+
+The purpose is to test the functionality of QoS remapping of tunnel traffic.
+
+## Scope
+
+The test is targeting a running SONiC system with fully functioning configuration.
+Purpose of the test is to verify a SONiC switch system correctly performs QoS remapping for tunnel traffic.
+
+## Testbed
+
+Supported topologies: dual-tor testbed
+
+## Setup configuration
+
+No setup pre-configuration is required, test will configure and return testbed to the initial state.
+
+## Test cases
+The test suite is categorized into two groups. 
+
+## Test case group for packet encapsulation
+### Setup of DUT switch
+1. Randomly select a ToR for testing, say `upper_tor`
+2. Randomly select a server (say ip `192.168.0.2`) facing interface on the selected ToR in step 1 (`upper_tor`), and toggle the mux status to `standby` in order to do packet encapsulation
+3. Startup `garp_service` on ptf to populate arp for servers
+### Test cases
+#### Test case 1 - Verify DSCP re-writing
+##### Test steps
+1. Generate packet with `DSCP = 3/4`, `target_ip = 192.168.0.2`, `src_ip = 1.1.1.1`
+2. Send the packets to `upper_tor` via a portchannel
+3. Verify the packets are encaped, and bounced back to T1
+4. Verify the `DSCP` value of bounced back packet is as expected (3->2, 4->6).
+
+#### Test case 2 - Verify traffic is egressed at expected queue
+##### Test steps
+1. Generate `100` packets with `DSCP = 3/4`, `target_ip = 192.168.0.2`, `src_ip = 1.1.1.1`
+2. Clear `queuecounter` with CLI `sonic-clear queuecounters`
+3. Send the packets to `upper_tor` via a portchannel
+4. Verify the packets are encaped, and bounced back to T1
+5. Verify the bounced back traffic is egressed at expected queue with CLI `show queue counter`. The packet counter for expected queue is supposed to be larger or equal to `100`. (`DSCP 3 -> Queue 2, DSCP 4 -> Queue 6`)
+
+
+## Test case group for packet decapsulation
+### Setup of DUT switch
+1. Swap `syncd` docker with `syncd-rpc` as `saithrift` call is required to do the validation
+2. Randomly select a ToR for testing, say `upper_tor`, and the unselected ToR would be `lower_tor`
+3. Randomly select a server (say ip `192.168.0.2`) facing interface on the selected ToR in step 1 (`upper_tor`), and toggle the mux status to `standby` in order to do packet encapsulation. server `192.168.0.2` would be active on `lower_tor`
+4. Startup `garp_service` on ptf to populate arp for servers
+
+### Test cases
+
+#### Test case 1 - Verify packets enter expected PG on lower_tor
+##### Test steps
+1. Generate `100` packets with `DSCP = 3/4`, `target_ip = 192.168.0.2`, `src_ip = 1.1.1.1`
+2. Send the `100` packets to `upper_tor` via a portchannel, and the encapped packets will be bounced to `lower_tor` via `T1`
```

Comment:
For our ptf setup, we need to send encapped packet (for eg. outer dscp: 2, inner dscp: 3) from T1 to active tor and verify that it is mapped to the correct pg (2)
As in other cases, please include dscps that map to other TC's as well and verify all others except lossless map to pg 0